frmts/ogcapi/gdalogcapidataset.cpp

```diff
@@ -2487,18 +2588,14 @@ OGRFeature *OGCAPITiledLayer::GetNextRawFeature()
             {
                 return nullptr;
             }
-            bool bEmptyContent = false;
-            m_poUnderlyingDS.reset(OpenTile(m_nCurX, m_nCurY, bEmptyContent));
-            if (bEmptyContent)
+
+            m_poUnderlyingDS.reset(OpenTile(m_nCurX, m_nCurY));
+            if (!m_poUnderlyingDS)
```

Comment:
why is the logic that distinguishes empty content from a failure to open a tile removed?

Reply:
Sorry, meant to point that out in the PR and then forgot.  The main reason was that it made for a more awkward api when moving `OpenTile` to `OGCAPIDataset` and my reading of the code was that it didn't make a difference.

So I just went back and read the original code to double check and see `bEmptyContent` used in 4 places:

1. line 2538:

```
    bool bEmptyContent = false;
    std::unique_ptr<GDALDataset> poUnderlyingDS(
        OpenTile(nX, nY, bEmptyContent));
```

`bEmptyContent` is not referenced in the rest of that method, so doesn't do anything.

2. Line 2567:

```
            bool bEmptyContent = false;
            m_poUnderlyingDS.reset(OpenTile(m_nCurX, m_nCurY, bEmptyContent));
            if (bEmptyContent || !m_poUnderlyingDS)
            {
                if (!IncrementTileIndices())
                    break;
                continue;
            }
```

In this case there is no distinction between returning a null pointer and `bEmptyContent  = true`

3. Line 2592:

```
                    bool bEmptyContent = false;
                    m_poUnderlyingDS.reset(
                        OpenTile(m_nCurX, m_nCurY, bEmptyContent));
                    if (bEmptyContent || !m_poUnderlyingDS)
                    {
                        continue;
                    }
```

In this case there is no distinction between returning a null pointer and `bEmptyContent  = true`

4. Line 2490:

```
bool bEmptyContent = false;
            m_poUnderlyingDS.reset(OpenTile(m_nCurX, m_nCurY, bEmptyContent));
            if (bEmptyContent)
            {
                if (!IncrementTileIndices())
                    return nullptr;
                continue;
            }
            if (m_poUnderlyingDS == nullptr)
            {
                return nullptr;
            }
```
In this case there is a difference. If `bEmptyContent` is true, then increment the indices, and loop again. Versus a nullptr in which case exist the method.

I collapsed those 2 together, so if a nullptr is returned then increment the indices and try again. 

The downside of that is if a server gives you an invalid url, you will check each tile before giving up. So one unsuccessful request could become many unsuccessful requests.

If that distinction is important then I can add `bEmptyContent`  back.

Reply:
I would prefer keeping the "failed to open" vs "nothing was downloaded" distinction, because when iterating over vector tiles, we want to go on silently if a tile is missing (ie call IncrementTileIndices()), but if we download something and can't GDALOpen() it, then we should just stop

Reply:
Ok. Suppose I can't just raise an exception?

Reply:
> Suppose I can't just raise an exception?

we tend to avoid them in the code base